src/main/scala/com/fulcrumgenomics/util/NcbiRefSeqGffSource.scala

```diff
@@ -0,0 +1,266 @@
+    */
+  def apply(lines: IterableOnce[String], includeXs: Boolean, dict: Option[SAMSequenceDictionary]): NcbiRefSeqGffSource =
+    NcbiRefSeqGffSource(lines.iterator, includeXs, dict)
+
+  /** Parses a RefSeq GFF file from an (optionally gzipped) text file.
+    *
+    * @param path the path to the file to parse
+    * @param includeXs whether to include experimental transcripts (i.e. XM_* XP_* and XR_*).
+    * @return an instance of [[NcbiRefSeqGffSource]] that can be queried for contents
+    */
+  def apply(path: FilePath, includeXs: Boolean, dict: Option[SAMSequenceDictionary]): NcbiRefSeqGffSource =
+    NcbiRefSeqGffSource(Io.readLines(path), includeXs, dict)
+}
+
+
+/** Queryable object that contains gene/transcript/exon information parsed from a RefSeq GFF file.
+  *
+  * @param lines the GFF lines to parse
+  * @param includeXs whether to include experimental transcripts (i.e. XM_* XP_* and XR_*).
+  * @param dict an optional sequence dictionary which, if provided, to use to limit the set of chromosomes
+  *             to process
+  */
+class NcbiRefSeqGffSource private(lines: Iterator[String],
+                                  val includeXs: Boolean,
+                                  val dict: Option[SAMSequenceDictionary] = None) extends Iterable[Gene] with LazyLogging {
+  import NcbiRefSeqGffSource._
+
+  private val chroms = new ArrayBuffer[Chrom]()
+  private val genes  = mutable.LinkedHashMap[String,Gene]()
+
+  /** NOTE about the structure of the RefSeq GFF files.
+    *
+    * Although GFF files are inherently flat with each line representing a record, it is best to
+    * think of RefSeq GFF files as hierarchical files with different kinds of records "nested"
+    * within each other.  The hierarchy looks something like:
+    *   * Chromosome region line
+    *       * Gene lines
+    *           * Transcript (and transcript-like entity) lines
+    *               * Exon lines
+    *               * CDS lines
+    *
+    * Thus the parsing takes place top-down by a) scanning for the next line of the expected type
+    * and then b) attempting to parse that line and it's children.
+    */
+
+  private val _bufferedLines = lines.bufferBetter
+
+  val genomeBuild: Option[String] = {
+    // Make an iterator of generic GFF records that is buffered so we can peek at the next rec
+    val headerLines = _bufferedLines.takeWhile(_.startsWith("#")).toIndexedSeq
+    headerLines.find(_.startsWith("#!genome-build")).map { l => l.split("""\s+""")(1) }
+  }
+
+  {
+    val iter = _bufferedLines.filter(l => !l.startsWith("#")).map { line =>
+      val fs = line.split('\t')
+      GffRecord(fs(0), fs(1), fs(2), int(fs(3)), int(fs(4)), fs(6) == "-", if (fs(7) == ".") -1 else int(fs(7)), fs(8))
+    }.bufferBetter
+
+    while (iter.nonEmpty) {
+      // Scan for the next chromosome record, discard any other records along the way
+      iter.find(rec => rec.kind == "region" && rec.accession.startsWith("NC_")) match {
```

Comment:
`startsWith("NC_")` is very specific. the gff spec says:
`a standard chromosome name or an Ensembl identifier such as a scaffold ID, without any additional content such as species or assembly`

would it make more sense to grab on record. parse the accession name for the region and then use that in a takeWhile instead of constructing the inner iter below?

Reply:
Ok ... I took a shot at allowing mapping of the other accessions using aliases in the sequence dictionary.  It means constructing a sequence dictionary that has the accessions as aliases, but is better I think.